pkg/engine/lifecycletest/analyzer_test.go

```diff
@@ -163,3 +165,167 @@ func TestSimpleAnalyzeStackFailure(t *testing.T) {
 	_, res := TestOp(Update).Run(project, p.GetTarget(t, nil), p.Options, false, p.BackendClient, nil)
 	assert.NotNil(t, res)
 }
+
+// TestResourceRemediation tests a very simple sequence of remediations. We register two, to ensure that
+// the remediations are applied in the order specified, an important part of the design.
+func TestResourceRemediation(t *testing.T) {
+	t.Parallel()
+
+	loaders := []*deploytest.PluginLoader{
+		deploytest.NewProviderLoader("pkgA", semver.MustParse("1.0.0"), func() (plugin.Provider, error) {
+			return &deploytest.Provider{}, nil
+		}),
+		deploytest.NewAnalyzerLoader("analyzerA", func(_ *plugin.PolicyAnalyzerOptions) (plugin.Analyzer, error) {
+			return &deploytest.Analyzer{
+				RemediateF: func(r plugin.AnalyzerResource) ([]plugin.Remediation, error) {
+					// Run two remediations to ensure they are applied in order.
+					return []plugin.Remediation{
+						{
+							PolicyName:        "ignored",
+							PolicyPackName:    "analyzerA",
+							PolicyPackVersion: "1.0.0",
+							Description:       "a remediation that gets ignored because it runs first",
+							Properties: resource.PropertyMap{
+								"a":   resource.NewStringProperty("nope"),
+								"ggg": resource.NewBoolProperty(true),
```

Comment:
I may be missing something about the test setup - but why do we not need to verify that that `ggg` is set to `true` by this remediation policy?

Reply:
It is explicitly overwritten by the second remediation.